File: app/src/main/java/fr/openium/examplepam/adapter/ContactAdapter.java
package fr.openium.examplepam.adapter;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import java.util.List;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;
import fr.openium.examplepam.R;

public class ContactAdapter extends RecyclerView.Adapter<ContactAdapter.ContactViewHolder> {
    private List<String> contacts;

    public ContactAdapter(List<String> contacts) {
        this.contacts = contacts;
    }

    @NonNull
    @Override
    public ContactViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(parent.getContext()).inflate(R.layout.item_contact, parent, false);
        return new ContactViewHolder(view);
    }

    @Override
    public void onBindViewHolder(@NonNull ContactViewHolder holder, int position) {
        holder.textViewItemContact.setText(contacts.get(position));
    }

    @Override
    public int getItemCount() {
        return contacts.size();
    }

    static class ContactViewHolder extends RecyclerView.ViewHolder {
        TextView textViewItemContact;

        ContactViewHolder(View view) {
            super(view);
            textViewItemContact = view.findViewById(R.id.textViewItemContact);
        }
    }
}

File: app/src/main/java/fr/openium/examplepam/adapter/ContactAdapter.kt
package fr.openium.examplepam.adapter

import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ImageButton
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import fr.openium.examplepam.R
import fr.openium.examplepam.adapter.ContactAdapter.ContactViewHolder
import fr.openium.examplepam.model.CallContact

class ContactAdapter(
    private val contacts: List<CallContact>,
    private val listener: OnContactClickedListener
) : RecyclerView.Adapter<ContactViewHolder>() {
    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ContactViewHolder {
        val view = LayoutInflater.from(parent.context).inflate(R.layout.item_contact, parent, false)
        return ContactViewHolder(view)
    }

    override fun onBindViewHolder(holder: ContactViewHolder, position: Int) {
        val contact = contacts[position]
        holder.textViewItemContact.text = contact.name
        holder.itemView.setOnClickListener {
            listener.onContactClicked(contact)
        }
        holder.imageButtonCall.setOnClickListener {
            listener.onCallClicked(contact)
        }
    }

    override fun getItemCount(): Int {
        return contacts.size
    }

    class ContactViewHolder(view: View) : RecyclerView.ViewHolder(view) {
        var textViewItemContact: TextView
        var imageButtonCall: ImageButton

        init {
            textViewItemContact = view.findViewById(R.id.textViewItemContact)
            imageButtonCall = view.findViewById(R.id.call)
        }
    }

    interface OnContactClickedListener {
        fun onContactClicked(contact: CallContact)
        fun onCallClicked(contact: CallContact)
    }
}